Presentation/PresentationStyle.swift

```diff
@@ -99,7 +99,28 @@ public extension PresentationStyle {
             return from.modallyPresentQueued(vc, options: options) {
                 Future { completion in
                     let bag = DisposeBag()
-                    bag += viewController.installDismissButton().onValue { completion(.failure(PresentError.dismissed)) }
+
+                    // The presentationController of an alert controller should not have its delegate modified
+                    if !(vc is UIAlertController) {
+                        let customPresentationDelegate: CustomAdaptivePresentationDelegate
+                        if let givenDelegate = viewController.customAdaptivePresentationDelegate {
+                            customPresentationDelegate = givenDelegate
+                        } else {
+                            customPresentationDelegate = CustomAdaptivePresentationDelegate()
+                            bag.hold(customPresentationDelegate)
+                        }
+
+                        customPresentationDelegate.allowSwipeDismissAlways = options.contains(.allowSwipeDismissAlways)
+                        vc.presentationController?.delegate = customPresentationDelegate
```

Comment:
Not sure I understand the memory leak issue. I was thinking something like this instead:

```swift
let delegate =  (vc.presentationController?.delegate as? CustomAdaptivePresentationDelegate) ?? CustomAdaptivePresentationDelegate()
vc.presentationController?.delegate = delegate
bag.hold(delegate)
bag += delagate.presentationControllerShouldDismiss.set {
  return Bool.... based on .allowSwipeDismissAlways etc
}

bag += delagate.didDismissSignal.onValue {
  completion(.failure(PresentError.dismissed))
}

```

Reply:
It's very weird leak, will try to understand more about it now. Basically seeing a memory leak alert (and a check in the memory graph confirms it) when running the UI tests when accessing `viewController.presentationController` (even when the rest of the logic is commented out). I think the problem is coming from the fact that [If you have not yet presented the current view controller, accessing presentationController creates a presentation controller](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621426-presentationcontroller). In our case `viewController` is not presented but if `vc` and `viewController` are different (when embedding in a navigation controller for example) then the one that will be presented and needs a presentation controller is `vc` and the presentation controller we just created for `viewController` leaks. Maybe related to http://www.openradar.appspot.com/36256721 . Will test this assumption now.

Reply:
Yep, that's what's happening. Here is a branch where you can reproduce it: https://github.com/iZettle/Presentation/tree/ios13-modal-sheets-leak. Not sure if there is a better workaround for a leak in UIKit. We can ofc add a comment about it...

<img width="940" alt="Screenshot 2019-07-25 at 12 42 08" src="https://user-images.githubusercontent.com/1555713/61868533-df5c1180-aed9-11e9-917a-2986e2c1d346.png">